main.go

```diff
@@ -176,15 +177,19 @@ func parsePath(path string) (ipath.Path, error) {
 		return nil, fmt.Errorf("%q could not be parsed: %s", path, err)
 	}
 
-	switch proto := u.Scheme; proto {
+	proto := u.Scheme
+	switch proto {
 	case "ipfs", "ipld", "ipns":
-		ipfsPath = ipath.New(gopath.Join("/", proto, u.Host, u.Path))
+		ipfsPath, err = ipath.NewPath(gopath.Join("/", proto, u.Host, u.Path))
 	case "http", "https":
-		ipfsPath = ipath.New(u.Path)
+		ipfsPath, err = ipath.NewPath(u.Path)
 	default:
 		return nil, fmt.Errorf("%q is not recognized as an IPFS path", path)
 	}
-	return ipfsPath, ipfsPath.IsValid()
+	if err != nil {
+		return nil, fmt.Errorf("cannot create %s path: %w", proto, err)
```

Comment:
This method is just parsing the path, so can we change it to:

```suggestion
		return nil, fmt.Errorf("cannot create well-formed IPFS path with proto '%s': %w", proto, err)
```

Reply:
There is a bigger problem: the change done here to use the new package causes paths without the `/ipfs/` prefix to no longer work.  I need to fix that first.

Reply:
I just return the error from `ipath.NewPath` in this case now.